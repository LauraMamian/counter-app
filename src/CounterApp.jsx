import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);
    
    function handleAdd() {
        setCounter(counter + 1);
        // setCounter((c) => c + 1);
    }

    function handleSubtract() {
        setCounter(counter - 1);
    }

    function handleReset() {
        setCounter(value);
    }

    return (
        <>
            <h1> Counter App </h1>
            <h2> {counter}</h2>

            <button onClick={handleAdd}> +1 </button>
            <button onClick={handleSubtract}> -1 </button>
            <button onClick={handleReset}> Reset </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}