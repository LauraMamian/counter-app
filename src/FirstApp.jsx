import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {

    return (
        <>
            {/* <code> { JSON.stringify(newMessage)} </code> */}
            <h1 data-testid="test-title"> {title} </h1>
            <p> {subTitle} </p>
            <p> {subTitle} </p>
            <p> {name} </p>
        </>
    )
}

FirstApp.propTypes = {
    subTitle: PropTypes.string,
    title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
    name: 'Laura Mamián',
    subTitle: 'No hay subtítulo',
    // title: 'No hay título',
}