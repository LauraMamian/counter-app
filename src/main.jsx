import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'

// import {HelloWordApp} from './HelloWordApp'
// import { FirstApp } from './FirstApp'
import { CounterApp } from './CounterApp'

function App() {
  return <div>Hello World</div>
}

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={4} />
  </React.StrictMode>
)