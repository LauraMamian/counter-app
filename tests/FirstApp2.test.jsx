import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {

    const title = 'Título de prueba';
    const subTitle = 'Subtítulo de prueba';

    test('Debe hacer match con el Snapshot', () => {
        const { container } = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrat el título "Esto es una prueba"', () => {

        render(<FirstApp title={title} />);
        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debe mostrar el título en un h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('Debe mostrar el subtítulo enviado por props', () => {
        render(
            <FirstApp
                title={title}
                subTitle={subTitle}
            />
        );

        expect(screen.getAllByText(subTitle).length).toBe(2);
    });
});