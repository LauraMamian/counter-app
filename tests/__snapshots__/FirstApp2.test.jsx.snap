// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Título de prueba
     
  </h1>
  <p>
     
    No hay subtítulo
     
  </p>
  <p>
     
    No hay subtítulo
     
  </p>
  <p>
     
    Laura Mamián
     
  </p>
</div>
`;
